code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1330,6 +1330,10 @@
 	if (affected_mob.get_timed_status_effect_duration(/datum/status_effect/hallucination) >= 10 SECONDS)
 		affected_mob.adjust_hallucinations(-10 SECONDS * REM * seconds_per_tick)
 
+	if(affected_mob.getStaminaLoss() >= 100)
+		for(var/datum/reagent/medicine/haloperidol/H in affected_mob.reagents.reagent_list)
```

Comment:
Does remove_reagent not handle trying to remove a chem that isnt there?

What is the sanity around this vs. just removing the reagent without looping through reagent list everytime? You already know it's in there because you're metabolizing it by the time you get to this proc.

Reply:
if(affected_mob.getStaminaLoss() >= 100)
		metabolization_rate = 1 * REAGENTS_METABOLISM
	else
		metabolization_rate = 0.4 * REAGENTS_METABOLISM

Reply:
Huh, i thought that would throw an error had i not included that. I tested it previously and ran into issues, testing again and somehow it got through. Okay, then.

Reply:
Eh, after removing that it's giving me errors again, guess i spoke too soon.

Reply:
I'll try altering it's metabolism instead of using remove_reagent.

Reply:
Reverted, you got any ideas that doesn't break it?